library/Theme/CustomModuleHelper.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Simrishamn\Theme;
+
+class CustomModuleHelper
+{
+    
+    const DOMAIN        = 'simrishamn';
+    const MODULE_FOLDER = SIMRISHAMN_PATH . '/custom-modules/';
+    
+    /**
+     * Initialize the Module
+     *
+     * @param object $object Current Class object
+     * @return object The module settings
+     */
+    public static function setModule ($object) : object
+    {
+        
+        if (!$object)
+            throw new \Exception('No object was found.');
+        
+        $obj = new \stdClass();
+        
+        $obj->moduleName    = self::moduleName($object);
+        
+        $obj->slug          = self::kebabcase($obj->moduleName);
+        $obj->fields        = self::fields($obj->moduleName);
+        
+        $obj->nameSingular  = self::translate($object->nameSingular);
+        $obj->namePlural    = self::translate($object->namePlural);
+        $obj->description   = self::translate($object->description);
+        
+        return $obj;
+        
+    }
+    
+    /**
+     * Current Class name
+     *
+     * @param string $object Current Class object
+     * @return string The module name
+     */
+    public static function moduleName ($object) : string
+    {
+        
+        return (new \ReflectionClass($object))->getShortName();
+        
+    }
+    
+    /**
+     * Creates a slugified kebabcase string
+     *
+     * @param string $string The string to be converted
+     * @return string The converted string
+     */
```

Comment:
Might want to add a reference to the StackOverflow answer since it's quite a few lines!